src/OMSimulatorLib/System.cpp

```diff
@@ -855,9 +848,12 @@ oms_status_enu_t oms3::System::addTLMBus(const oms3::ComRef &cref, const std::st
   if(subsystem != subsystems.end()) {
     return subsystem->second->addTLMBus(tail, domain, dimensions, interpolation);
   }
-  auto externalmodel = externalmodels.find(head);
-  if(externalmodel != externalmodels.end()) {
-    return externalmodel->second->addTLMBus(tail, domain, dimensions, interpolation);
+  auto externalmodel = components.find(head);
+  if(externalmodel != components.end())
+  {
+    if (oms_component_external != externalmodel->second->getType())
+      return logError_OnlyForTlmSystem;
```

Comment:
Perhaps this should be a new error called logError_OnlyForExternalModels?

Reply:
I will change it. However, we have to improve most of the error messages, that do not reference a model/system/component to make them more meaningful.